godump_test.go

```diff
@@ -891,6 +869,27 @@ func TestDumpWithCustomWriter(t *testing.T) {
 	}
 }
 
+func TestDumpWithCustomSkipStackFrames(t *testing.T) {
+	// caller stack frames are
+	//	1	godump.go           github.com/goforj/godump.findFirstNonInternalFrame			skip by initialCallerSkip
+	//	2	godump.go           github.com/goforj/godump.printDumpHeader					skip by initialCallerSkip
+	//	3	godump.go           github.com/goforj/godump.(*Dumper).DumpStr					skip by fail names contain godump.go
+	//	4	godump_test.go      github.com/goforj/godump.TestDumpWithCustomSkipStackFrames
+	//	5	testing.go          testing.tRunner
+	out := NewDumper().DumpStr("test")
+	assert.Contains(t, out, "godump_test.go")
+
+	out = NewDumper(WithSkipStackFrames(1)).DumpStr("test")
+	assert.NotContains(t, out, "godump_test.go")
+	assert.Contains(t, out, "testing.go")
+
+	wrappedDumpStr := func(v any) string {
+		return NewDumper(WithSkipStackFrames(1)).DumpStr(v)
+	}
+	out = wrappedDumpStr("test")
+	assert.Contains(t, out, "godump_test.go:889")
```

Comment:
Unfortunately, this is not maintenable.

Move out the wrapper in a dedicated function in the test file.

Check for its name in the message.

A good way to test this would be to wrap it twice and check that skipping no frames leads to detection of the deepest wrapper, and skipping one would lead to the first one.